### PR TITLE
Support VS2015, VC14, and MsBuild14

### PR DESCRIPTION
**Pull request by shoelzer**
This is a redo/update of my failed PR #2232. Hopefully my notifications don't fall into a black hole this time.

### CODE REVIEW

**Comment by sa2ajj** [comment]
I'd like to repeat my suggestion: instead of `'...\\...'` use `r'...\...'`  This improves readability.

**Comment by shoelzer** [comment]
@sa2ajj I already did that for the code that I touched. Did you mean to convert all strings in the file?

**Comment by sa2ajj** [comment]
If you could do that, that'd be great!

On a unrelated note: you added a number of steps, however they are not present in `setup.py`: so they can't be used via plug-in infrastructure.

**Comment by shoelzer** [comment]
I didn't know about setup.py. I just pushed an update for that and string literals everywhere in vstudio.py.

**Comment by shoelzer** [comment]
@sa2ajj Can this get merged now?

File: master/buildbot/steps/vstudio.py [hunk 1]
```diff
@@ -198,27 +198,27 @@ def finished(self, result):
 
 class VC6(VisualStudio):
 
-    default_installdir = 'C:\\Program Files\\Microsoft Visual Studio'
+    default_installdir = r'C:\Program Files\Microsoft Visual Studio'
```
Comment (Stibbons) [inline master/buildbot/steps/vstudio.py]: why r"" ? doesn't it parse "\P" as "P", and not the character "\" + "P" ?

Comment (shoelzer) [inline master/buildbot/steps/vstudio.py]: I used r"" as @sa2ajj requested so that all the slashes in path strings can be single instead of double.

**Comment by shoelzer** [comment]
@sa2ajj Can this get merged now?

**Comment by shoelzer** [comment]
It's been two months. Anybody there?

**Comment by sa2ajj** [comment]
@shoelzer I'm sorry, it took so long (I've been a bit too busy).

**Pull request merged**